test/test_basic_extractor.py

```diff
@@ -0,0 +1,171 @@
+"""
+Test the code in rnnalayse.extractors.base_extractor.py.
+"""
+
+import itertools
+import unittest
+from unittest.mock import patch, MagicMock
+
+import numpy as np
+from overrides import overrides
+import torch
+
+from rnnalyse.extractors.base_extractor import Extractor
+from rnnalyse.models.language_model import LanguageModel
+from test_utils import create_sentence_dummy_activations
+
+
+# GLOBALS
+ACTIVATION_DIM = 10
+ACTIVATION_NAMES = [(0, "hx"), (0, "cx")]
+ACTIVATIONS_DIR = "test_data"
+
+
+class MockLanguageModel(LanguageModel):
+    """
+    Create a Mock version of the LanguageModel class which returns pre-defined dummy activations.
+    """
+    def __init__(self, num_layers, hidden_size, all_tokens, all_activations):
+        super().__init__()
+        self.num_layers = num_layers
+        self.hidden_size = hidden_size
+        self.all_tokens = all_tokens
+        self.all_activations = all_activations
+        self.all_pairs = None
+        self.reset()
+
+    @overrides
+    def forward(self, token, activations):
+        # Consume next activation, make sure it's the right token
+        next_token, next_activation = next(self.all_pairs)
+        assert token == next_token
+
+        return None, {0: {"hx": next_activation, "cx": next_activation}}
+
+    def reset(self):
+        """ Reset the activations for next test. """
+        self.all_pairs = zip(self.all_tokens, self.all_activations)
+
+
+class TestExtractor(unittest.TestCase):
```

Comment:
maybe we can also add a test that tests if the pickle functions are called with the right arguments?

Reply:
Done!